### ISSUE TITLE
Fix "Only X left" functionality

### ISSUE THREAD

**Issue by nmalevanec** [magento/inventory#1165]
### Preconditions
1. Magento with MSI 'Single Stock' mode

### Steps to reproduce
1. Run Magento\Catalog\Test\TestCase\Product\CreateSimpleProductEntityTest::CreateSimpleProductEntityTestVariation10
### Expected result
1. Test should pass successfully.

### Actual result
1. Test fails.
```
Product inventory threshold message display is not correct.
Failed asserting that true matches expected false.

/var/www/html/dev/tests/functional/tests/app/Magento/Catalog/Test/Constraint/AssertProductInventoryThreshold.php:50
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/ObjectManager/Factory.php:57
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/ObjectManager.php:81
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/Constraint/AbstractConstraint.php:155
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/Constraint/Composite.php:109
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/TestCase/Injectable.php:231
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/TestCase/Functional.php:300
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/TestCase/Functional.php:217
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/TestCase/Injectable.php:210
/var/www/html/dev/tests/functional/vendor/magento/mtf/Magento/Mtf/TestCase/Injectable.php:166
```

### Test steps
1. Navigate to admin area.
2. Create simple product with qty = 10
3. Navigate to Stores > Configuration > Catalog > Inventory > Stock Options
4. Set Only X left Threshold to 5.
5. Navigate to storefront.
6. Register and login with customer.
7. Add 5 pieces of product to cart and proceed to checkout.
8. Place order.
9. Navigate to product view page.
11. Test expects "Only 5 left" message appears on product view page.

### Additional Info
Standard magento behaviour -  "Only 5 left" message appears immediately after place order, but. On MSI "Only 5 left" message appears only after shipment created.
Additional steps with shipment placement should be added into test before assertions.

**Comment by aleron75** [comment]
Just submitted PR #1348 to address this issue.

Working on this with @maghamed we realized that it could be a good idea to refactor the `\Magento\InventorySales\Model\GetProductSalableQtyInterface` to take the min qty (a.k.a. "stock safety threshold") into account. 

This value affects the salability and indeed in several circumstances we have to subtract the min qty from the salable qty.

Furthermore, we considered it could be useful for merchants to see in the "real" salable quantity in the product admin grid.

We can open another issue to address this refactoring that shouldn't introduce logical changes.

**Comment by smoskaluk** [comment]
@aleron75 Unfortunatly, the problem is not fixed. Please re-check

**Comment by smoskaluk** [comment]
@aleron75 @maghamed I could not verify this PR because message 'Only x left' appear on product page only after flushing the cache. Please re-check
